tests/test-socket-frag.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# This test exercises message fragmentation over a socket by setting the
+# maximum transmission unit size to a very low value, then passing a (very large)
+# DNSSEC query/response over the socket.  If fragmentation is successful, we will
+# have an identical file on the other side!
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+infile=$script_dir/private-tests/dnssec.pcap
+outfile_frag=/tmp/testfrag.sock.pres
+outfile=/tmp/testfrag.pres
+sockaddr=127.0.0.1/8080
+entry_count=1
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+echo "Testing datagram socket fragmentation and reassembly: "
+
+# Write the input file to an nmsg file which will be diffed later.
+$nmsgtool_test -ddddd -c $entry_count -p $infile -V base -T dnsqr -o $outfile
+
+# Create a listener, wait a second, then create a writer on the same IP/socket.
+$nmsgtool_test -ddddd --unbuffered -m 1 -c $entry_count -o $outfile_frag -l $sockaddr &
+listener_pid=$!
+sleep 1
+$nmsgtool_test -ddddd --unbuffered -m 1 -c $entry_count -p $infile -V base -T dnsqr -s $sockaddr &
+writer_pid=$!
+echo Listening PID: $listener_pid, writing PID: $writer_pid
+
+# Wait for the messages to send/receive.
+sleep 1
+
+# Kill em all!
+kill $writer_pid >/dev/null 2>&1
+killwriter_pid=$!
+kill $listener_pid >/dev/null 2>&1
+killlistener_pid=$!
+wait $killwriter_pid
```

Comment:
You should just be able to wait on `$writer_pid` and `$listener_pid` directly...

Reply:
Also, are you sure we can't just wait on them instead of killing them?

Reply:
We can (and now do) wait on them instead of killing them.